src/index.coffee

```diff
@@ -94,8 +115,31 @@ reinstall_all = (options = {}, packages) ->
 
   return if packages then curried packages else curried
 
+shrinkwrap = (packages, options = {}) ->
+  shrinkwrap_json =
+    dependencies: {}
+
+  for pkg in packages
+    [ whole, git_at_github, name, branch] = pkg.match /^(.+?):(.+?)(?:\.git)(?:#(.+))?$/
+    ref = "refs/heads/#{branch}"
```

Comment:
Will this work if a sha is specified in package.json instead of a branch, like in a scenario described in lzrski/npm-git-install#7 issue? It's probably an edge-case, because user will most likely not use shrinkwrap if she already has a fixed sha in package.json. Nevertheless it would be nice to have it work in either case.